### ISSUE TITLE
support `data_set_external_id` in gh-action

### ISSUE THREAD

**Issue by spex66** [cognitedata/function-action-oidc#35]
Should not be a big deal to add support for logical dataset-ids aka "external-ids".
```
data_set_id = client.data_sets.retrieve(id=data_set_external_id).id
```

Which allows configuration to be environment agnostic

**Comment by haakonvt** [comment]
Yes and no - if I remember correctly; it has to do with how we verify the necessary capabilities for the user; `groups.list` returns scope for the files-related capabilities _only_ with data set ID(s) (not XID). The way I see it we could skip the checks of file capabilities in `access.py` - and rather just fail later when trying to upload the zipped function file object?

For reference:
https://github.com/cognitedata/function-action-oidc/pull/3

API file upload does not support data set external id when uploading file:
https://docs.cognite.com/api/v1/#operation/initFileUpload

...and so the same goes for the SDK:
https://github.com/cognitedata/cognite-sdk-python/blob/82069d0984150bdbc66fd6b87a381e5656d3870e/cognite/client/_api/files.py#L594

This means by allowing data set XID we would require an additional capability; `Dataset:READ`, just to convert from given XID to ID in order to be able to upload.

🤔 

😢

**Comment by qTipTip** [comment]
Hello! 

Closing this issue as this action is now deprecated. 
This action's functionality is now offered by the CDF Toolkit: 

https://docs.cognite.com/cdf/deploy/cdf_toolkit/references/resource_library/#functions
